Tests/Common/Securities/Futures/FuturesExpiryFunctionsTests.cs

```diff
@@ -16,12 +16,114 @@
 using System;
 using NUnit.Framework;
 using QuantConnect.Securities.Future;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+using System.Xml.Serialization;
+using System.IO;
 
 namespace QuantConnect.Tests.Common.Securities.Futures
 {
     [TestFixture]
     public class FuturesExpiryFunctionsTests
     {
+        /// <summary>
+        /// Dates for Termination Conditions of futures
+        /// </summary>
+        public class Dates
+        {
+            public DateTime contractMonth;
+            public DateTime lastTrade;
+            public Dates() { }
+            public Dates(DateTime c, DateTime l)
+            {
+                contractMonth = c;
+                lastTrade = l;
+            }
+        }
+
+        /// <summary>
+        /// Symbol and list of dates for testing Futures
+        /// </summary>
+        public class SymbolData
+        {
+            public String symbol;
+            public List<Dates> dateList;
+            public SymbolData() { }
+            public SymbolData(String symbol, List<Dates> list)
+            {
+                this.symbol = symbol;
+                this.dateList = list;
+            }
+        }
+
+        [Test]
+        public void TestAllExpiryDateFunctions()
```

Comment:
I think it would be best to have a separate test for each ExpiryDateFunction. That way, in two years from now when someone makes a minor tweak and these tests start to fail, it's easier to identify which expiry date function is the problem.